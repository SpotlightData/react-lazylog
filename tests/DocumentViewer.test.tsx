import * as React from 'react';
import { render, fireEvent, cleanup, waitForElement } from 'react-testing-library';
import { DocumentViewer } from '@spotlightdata/document-viewer';

const text = `
Lorem ipsum dolor sit amet, 
consectetur adipiscing elit. 
Nunc semper sagittis libero sit amet venenatis. Curabitur
nec sem condimentum, volutpat sapien nec, pulvinar felis.
Integer orci tellus, bibendum id condimentum sed, tincidunt ac urna.
Integer vehicula vel eros ac pulvinar. Sed accumsan lacus purus
`.split('\n');

describe('DocumentViever', () => {
  it('should render', () => {
    const { container } = render(
      <DocumentViewer width={200} height={200} text={text} rulerId="myId" />
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should create a ruler', () => {
    const rulerId = 'myId';
    const { debug } = render(
      <DocumentViewer width={200} height={200} text={text} rulerId={rulerId} />
    );
    expect(document.querySelector(`#${rulerId}`)).toBeInTheDocument();
  });

  it('should clean up the ruler', () => {
    const rulerId = 'myId';
    const { unmount, container } = render(
      <DocumentViewer width={200} height={200} text={text} rulerId={rulerId} />
    );
    unmount();
    expect(container.querySelector(`#${rulerId}`)).not.toBeInTheDocument();
  });
});
