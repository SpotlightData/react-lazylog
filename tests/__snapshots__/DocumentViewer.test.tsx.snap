// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentViever should render 1`] = `"<div id=\\"nBi6IrbyE\\"><div style=\\"position: relative; height: 200px; width: 200px; overflow: auto; will-change: transform; direction: ltr;\\"><div style=\\"height: 0px; width: 100%;\\"><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">0</span></div><div style=\\"display: inline-block;\\"></div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">1</span></div><div style=\\"display: inline-block;\\">Lorem ipsum dolor sit amet, </div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">2</span></div><div style=\\"display: inline-block;\\">consectetur adipiscing elit. </div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">3</span></div><div style=\\"display: inline-block;\\">Nunc semper sagittis libero sit amet venenatis. Curabitur</div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">4</span></div><div style=\\"display: inline-block;\\">nec sem condimentum, volutpat sapien nec, pulvinar felis.</div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">5</span></div><div style=\\"display: inline-block;\\">Integer orci tellus, bibendum id condimentum sed, tincidunt ac urna.</div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">6</span></div><div style=\\"display: inline-block;\\">Integer vehicula vel eros ac pulvinar. Sed accumsan lacus purus</div></div><div style=\\"position: absolute; left: 0px; top: 0px; height: 0px; width: 100%; display: flex;\\"><div style=\\"display: inline-block; height: 100%; width: 30px;\\"><span style=\\"float: right; padding: 0px 0.5em 0px 0px;\\">7</span></div><div style=\\"display: inline-block;\\"></div></div></div></div></div>"`;
